contracts/Cryptoxo.sol

```diff
@@ -0,0 +1,254 @@
+                //Alert(oxoBoard[i][0], oxoBoard[i][1], oxoBoard[i][2]);
+                UserWin(msg.sender, this.balance, "triggered by compareHorizontal()");
+                returnEtherToWinner();
+                return true;
+            }
+        }
+    }
+    
+    function compareVertical() public returns (bool){
+        for(uint8 i = 0; i <= 2; i++){
+            if(keccak256(oxoBoard[0][i]) == keccak256("-")) continue;
+            if(keccak256(oxoBoard[0][i]) == keccak256(oxoBoard[1][i]) && keccak256(oxoBoard[1][i]) == keccak256(oxoBoard[2][i])){
+                UserWin(msg.sender, this.balance, "triggered by compareVertical()");
+                returnEtherToWinner();
+                return true;
+            }
+        }
+    }
+    
+    function compareDiagonal() public returns (bool){
+        if(keccak256(oxoBoard[1][1]) == keccak256("-")) return false;
+        if(keccak256(oxoBoard[0][0]) == keccak256(oxoBoard[1][1]) && keccak256(oxoBoard[1][1]) == keccak256(oxoBoard[2][2])){
+            UserWin(msg.sender, this.balance, "triggered by compareDiagonal()");
+            returnEtherToWinner();
+            return true;
+        } else if(keccak256(oxoBoard[0][2]) == keccak256(oxoBoard[1][1]) && keccak256(oxoBoard[1][1]) == keccak256(oxoBoard[2][0])){
+            UserWin(msg.sender, this.balance, "triggered by compareDiagonal()");
+            returnEtherToWinner();
+            return true;
+        }
+    }
+    
+    function returnEtherToWinner() internal returns (bool){
+        msg.sender.transfer(this.balance);
+        return returnToInitialState();
+    }
+    
+    function returnEtherDraw() internal returns (bool) {
+        GameIsDraw("the game is draw", this.balance / 2, "goes to:", players.x, players.o);
+        
+        players.x.transfer(this.balance / 2);
+        players.o.transfer(this.balance);
+        
+        return returnToInitialState();
+    }
+    
+    function seeContractBallance() public returns (uint){
+        return this.balance;
+    }
+    
+    function showOxoBoard() public  {
+        showOxoBoardString("row 1:", string(oxoBoard[0][0]), string(oxoBoard[1][0]), string(oxoBoard[2][0]));
+        showOxoBoardString("row 2:", string(oxoBoard[0][1]), string(oxoBoard[1][1]), string(oxoBoard[2][1]));
+        showOxoBoardString("row 3:", string(oxoBoard[0][2]), string(oxoBoard[1][2]), string(oxoBoard[2][2]));
+    }
+    
+    
+    
+    function returnToInitialState() internal returns (bool){
+        oxoBoard[0] = ["-", "-", "-"];
+        oxoBoard[1] = ["-", "-", "-"];
+        oxoBoard[2] = ["-", "-", "-"];
+        
+        // I still have a question is this the right thing to do
+        // to simply equal values to zero
+        // or I should do some erase or something
+        // but no time for searches at the moment, will dig in later
+        playersVal[players.x] = "";
```

Comment:
if you want here to be zero you should set zero.

Reply:
zero equls to empty value, but I was wandersing if it's same in gas amount to equal to zero or to erase tha data. will check this one more when I come back